### ISSUE TITLE
Request: bumping `inference` extra to pydantic v2

### ISSUE THREAD

**Issue by jamesbraza** [huggingface/huggingface_hub#1726]
**Is your feature request related to a problem? Please describe.**

The `inference` extra requires pydantic v1 for `text-generation-inference`: https://github.com/huggingface/huggingface_hub/blob/v0.18.0/setup.py#L33

However, thanks to https://github.com/huggingface/text-generation-inference/pull/900, released in https://github.com/huggingface/text-generation-inference/releases/tag/v1.1.0, pydantic v2 is now supported there.

So this means `inference` extra can now support pydantic v2 as well! 🥳 

**Describe the solution you'd like**

Allowing pydantic v2 with Hugging Face's Inference API client.

**Describe alternatives you've considered**

n/a

**Additional context**

n/a

**Comment by Wauplin** [comment]
Hi @jamesbraza thanks for the suggestion! In general, I'd be in favor of relaxing the dependency version :)

Just to be sure, what does that imply in term of backward-compatibility? (I would prefer the `InferenceClient` to stay compatible with an earlier version of TGI as far as possible).

**Comment by jamesbraza** [comment]
The source code was compatible, but actually the test suite wasn't.  I thus opened two PRs to make the test suite compatible:

- https://github.com/huggingface/huggingface_hub/pull/1731
- https://github.com/huggingface/huggingface_hub/pull/1732

**Comment by Wauplin** [comment]
Thanks for opening PRs fixing these @jamesbraza!

It's too bad that I don't remember why I've pinned pydantic version in the first place :confused: (I did [a commit on main](https://github.com/huggingface/huggingface_hub/commit/2940a65b22e9552b0dd40f0b61f502f66896d46d) without description and bad practices always hits you back :sob:). Maybe it was because of the webhook payload test you fixed but I'm not sure. I'll check and make a few tests with pydantic 1.0 vs 2.0 just to be sure it works as expected.
